granary/flickr.py

```diff
@@ -375,13 +535,68 @@ def get_user_image(self, farm, server, author):
     return u'https://farm{}.staticflickr.com/{}/buddyicons/{}.jpg'.format(
       farm, server, author)
 
-  def user_url(self, handle):
-    """Convert a user's screen name to their Flickr profile page URL.
+  def user_id(self):
+    """Get the nsid of the currently authorized user. The first time this
+    is called, it will invoke the flickr.people.getLimits api method.
+
+    https://www.flickr.com/services/api/flickr.people.getLimits.html
+
+    Return:
+      a string
+    """
+    if not self._user_id:
+      resp = self.call_api_method('flickr.people.getLimits')
+      self._user_id = resp.get('person', {}).get('nsid')
+    return self._user_id
+
+  def path_alias(self):
+    """Get the path_alias of the currently authorized user. The first time this
+    is called, it will invoke the flickr.people.getInfo api method.
+
+    https://www.flickr.com/services/api/flickr.people.getInfo.html
+
+    Return:
+      a string
+    """
+    if not self._path_alias:
+      resp = self.call_api_method('flickr.people.getInfo', {
+        'user_id': self.user_id(),
+      })
+      self._path_alias = resp.get('person', {}).get('path_alias')
+    return self._path_alias
+
+  def user_url(self, user_id, path_alias):
+    """Convert a user's path_alias to their Flickr profile page URL.
 
     Args:
-      handle: string, the Flickr user's screen name
+      user_id (string): user's alphanumeric nsid
+      path_alias (string): if given, user's path_alias replaces the user_id
+        in canonical urls
 
     Returns:
       string, a profile URL
     """
-    return handle and 'https://www.flickr.com/people/%s/' % handle
+    return path_alias and 'https://www.flickr.com/people/%s/' % path_alias
+
+  def photo_url(self, user_id, path_alias, photo_id):
```

Comment:
I considered it, but usually it's constructing permalinks for an arbitrary user rather than the currently logged in user... Making them optional and defaulting to the current user felt slightly more confusing than it was worth.